src/main/scala/dxWDL/base/Extras.scala

```diff
@@ -618,11 +618,74 @@ object Extras {
         val reorgInput: String = checkedParseStringFieldReplaceNull(fields, "inputs") match {
             case None => throw new IllegalArgumentException(
                 "inputs must be specified in the custom_reorg section. " +
-                "Please set the value to null if there is no input."
+                  "Please set the value to null if there is no input."
             )
             case Some(x) => x
         }
 
+        return (reorgAppId,reorgInput)
+
+    }
+
+    def verifyReorgAppHasAccess(appDescribe: DXApplet.Describe, reorgAppId: String) = {
+
+        // check applet has access to the projet
+        val accessJson = appDescribe.getAccess()
+        val projectAccess = accessJson.get("project")
+        val access: String = if ( projectAccess != null) {
+
+            projectAccess.toString.replace("\"", "")
+        }
+        else ""
+
+        if ( access != "CONTRIBUTE" && access != "ADMINISTER" ) {
+
+            throw new PermissionDeniedException(s"ERROR: Applet for custom reorg stage ${reorgAppId } does not " +
+              s"have CONTRIBUTOR or ADMINISTRATOR access and this is required.", -1)
+
+        }
+
+    }
+
+    def veryifyRorgApp(reorgAppId: String) = {
+
+        // if reorgAppId is invalid, DXApplet.getInstance will throw an IllegalArgumentException
+        val app: DXApplet = DXApplet.getInstance(reorgAppId)
+        // if reorgAppId cannot be found, describe() will throw a ResourceNotFoundException
+        val appDescribe: DXApplet.Describe = app.describe()
+        // verify reorg app has at least contribute access
+        verifyReorgAppHasAccess(appDescribe, reorgAppId)
+
+    }
+
+    def verifyInputs(reorgInput: String) = {
+
+        // if provided, check that the fileID is valid and present
+        if ( reorgInput != "" ) {
+            // format dx file ID
+            val reorgFileID: String = reorgInput.replace("dx://", "")
+            // if input file  ID is invalid, DXFile.getInstance will thown an IllegalArgumentException
+            val file: DXFile = DXFile.getInstance(reorgFileID)
+            // if reorgFileID cannot be found, describe will throw a ResourceNotFoundException
+            val fileDescribe: DXFile.Describe = file.describe()
```

Comment:
No need to assign it to a value if it is never used.

Reply:
done